atc/api/artifactserver/create.go

```diff
@@ -20,47 +20,20 @@ func (s *Server) CreateArtifact(team db.Team) http.Handler {
 		// which we can lookup in the checksum field
 		// that way we don't have to create another volume.
 
-		workerSpec := worker.WorkerSpec{
+		workerSpec := worker.Spec{
 			TeamID:   team.ID(),
 			Platform: r.FormValue("platform"),
 			Tags:     r.Form["tags"],
 		}
 
-		volumeSpec := worker.VolumeSpec{
-			Strategy: baggageclaim.EmptyStrategy{},
-		}
-
-		volume, err := s.workerPool.CreateVolume(hLog, volumeSpec, workerSpec, db.VolumeTypeArtifact)
+		volume, artifact, err := s.workerPool.CreateVolumeForArtifact(hLog, workerSpec)
 		if err != nil {
 			hLog.Error("failed-to-create-volume", err)
 			w.WriteHeader(http.StatusInternalServerError)
 			return
 		}
 
-		// NOTE: there's a race condition here between when the
-		// volume gets created and when the artifact gets initialized
-
-		// Within this timeframe there's a chance that the volume could
-		// get garbage collected out from under us.
-
-		// This happens because CreateVolume returns a 'created' instead
-		// of 'creating' volume.
-
-		// In the long run CreateVolume should probably return a 'creating'
-		// volume, but there are other changes needed in FindOrCreateContainer
-		// with the way we create volumes for a container
-
-		// I think leaving the race condition is fine for now. Worst case
-		// is a fly execute will fail and the user will need to rerun it.
```

Comment:
I don't think so, because (at least based on this comment block) creating volumes don't get GC'd, and we only mark the volume as created after it's been initialized

Reply:
Ah, missed the `CreateVolume` returning a `creatingVolume` part